lib/Analysis/Allocation.cpp

```diff
@@ -396,6 +432,13 @@ class AllocationAnalysis {
       buffers.emplace_back(bufferIter.first);
     }
 
+    // Sort buffers by size in descending order to reduce the fragmentation
+    // of big buffers. Big buffers have a higher chance to overlap with multiple
+    // other buffers, and allocating them first (by calculateStarts) ensures a
+    // higher chance that they will occupy a standalone smem slot.
+    llvm::sort(buffers,
```

Comment:
nit: a stable sort may be more predictable/easy to reason about

Reply:
I'm not super clear about the comments. My understanding is that big buffers suffer from fragmentation caused by small buffers, but not big buffers cause fragmentation. Maybe you can reword the comments if my understanding is correct. @htyu

Reply:
That's more accurate. Thanks for the suggestion.